### PR TITLE
Parse parenthesized expressions

### PR DESCRIPTION
**Pull request by gittywithexcitement**
Also document things in ast.rs

### CODE REVIEW

File: src/ast.rs [hunk 1]
```diff
@@ -147,7 +149,9 @@ pub struct Usize(pub u32);
 /// See <https://google.github.io/xls/dslx_reference/#bit-type>
 #[derive(Debug, PartialEq, Clone)]
 pub struct RawBitType {
+    /// a bit type is differentiated on if it's signed or unsigned
```
Comment (RobSpringer) [inline src/ast.rs:152]: Comment is less complete-sentence-y than others. Capitalize, punctuate, etc.

File: src/lib.rs [hunk 2]
```diff
@@ -1223,28 +1230,28 @@ mod tests {
         let _ = expression_is_literal(*rhs);
     }
 
+    // Is the operation that evalutes first in the LHS or RHS of the second operation?
+    enum FirstsLocation {
```
Comment (RobSpringer) [inline src/lib.rs:1234]: I think this is a simple concept, but for some reason my feeble mind rebelling. Can you sketch an example here of both?

Comment (RobSpringer) [reply src/lib.rs:1234]: Ah - this is test code. Less critical...but I still wanna know :)

**Review by RobSpringer — APPROVED** [review]
As good a time as any: some part of my mind gets twisted up when files get long, even if they're split into functionality and tests. WDYT about splitting test cases into their own file? Is that a no-no in Rust? I'm open to the possibility that I just need to evolve; that it's bad practice...in practice. :)

Aside from that, LGTM! Feel free to merge regardless of the above discussion; it can happen offline 

or not at all

File: src/ast.rs (hunk 1 above)
Comment (gittywithexcitement) [reply src/ast.rs:152]: done

**Comment by gittywithexcitement** [comment]
> As good a time as any: some part of my mind gets twisted up when files get long, even if they're split into functionality and tests. WDYT about splitting test cases into their own file? Is that a no-no in Rust? I'm open to the possibility that I just need to evolve; that it's bad practice...in practice. :)
> 
> Aside from that, LGTM! Feel free to merge regardless of the above discussion; it can happen offline
> 
> or not at all

Sounds good. I'll do this in a commit that's at the end of my current commit chain, to avoid really painful rebases.

File: src/lib.rs (hunk 2 above)
Comment (gittywithexcitement) [reply src/lib.rs:1234]: scroll down to see usages

**Pull request merged**